### PR TITLE
Fixes pills having a delay.

### LINKED ISSUE #16433: Pills have a delay for self application
[Here](https://github.com/tgstation/-tg-station/blob/master/code/modules/reagents/reagent_containers/pill.dm#L12) it gives 0 delay for self applications(which is called in line 32 in the same file), but when you try to swallow a pill(given by yourself) you get a wait time.

### PR DESCRIPTION
**Pull request by Supermichael777**
TESTING AM I RIGHT?

aparently what i did doesnt work with values below like 20 so ANOTHER VAR
fixes #16433

### CODE REVIEW

File: code/modules/reagents/reagent_containers/pill.dm [hunk 1]
```diff
@@ -29,8 +30,9 @@
 
 	if(M == user)
 		M.visible_message("<span class='notice'>[user] attempts to [apply_method] [src].</span>")
-		if(!do_mob(user, M, self_delay))
-			return 0
+		if(!no_delay)
```
Comment (KorPhaeron) [inline code/modules/reagents/reagent_containers/pill.dm]: why not just if(self_delay)

Comment (Supermichael777) [inline code/modules/reagents/reagent_containers/pill.dm]: i was tired and didn't think about it

**Comment by Supermichael777** [comment]
FINE I'll change it

**Comment by Supermichael777** [comment]
@KorPhaeron

File: code/modules/reagents/reagent_containers/pill.dm [hunk 2]
```diff
@@ -9,8 +9,9 @@
 	var/apply_type = INGEST
 	var/apply_method = "swallow"
 	var/roundstart = 0
-	var/self_delay = 0 //pills are instant, this is because patches inheret their aplication from pills
-
+	var/self_delay = 1	//pills are instant, this is because patches inheret their aplication from pills This low a value doesnt wor for some reason
+	var/has_self_delay = 0	//crapy workaround is crapy
```
Comment (duncathan) [inline code/modules/reagents/reagent_containers/pill.dm]: Why are there two vars???

**Pull request closed without merging**